### PR TITLE
Pass to remove the memory space attribute from all types

### PR DESCRIPTION
**Pull request by newling**
Remove memory space annotation from all types.

    For example, replace
      ` %alloc = memref.alloc() : memref<1x1x8x4x8x4xbf16, 2 : i32> `
    with
      ` %alloc = memref.alloc() : memref<1x1x8x4x8x4xbf16> `

    and
      `  aie.objectfifo @obj0(%tile_0_0, {%tile_0_1}, 2 : i32) : !aie.objectfifo<memref<2048xbf16, 1 : i32>> `
    with
      `  aie.objectfifo @obj0(%tile_0_0, {%tile_0_1}, 2 : i32) : !aie.objectfifo<memref<2048xbf16>> `

    etc.

    The memory space annotation is not required or even supported after lowering
    from the AMDAIE dialect to the AIE dialect. The memory space annotations
    are introduced during the tiling of linalg operations, and so this pass
    should be run somewhere between tiling and lowering to the aie dialect.

### CODE REVIEW

File: compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/AMDAIERemoveMemorySpace.cpp
Comment (jtuyls) [inline compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/AMDAIERemoveMemorySpace.cpp:134]: Why is this needed for `ObjectFifoCreateOp`?

Comment (newling) [reply compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/AMDAIERemoveMemorySpace.cpp:138]: I added a comment in the code answering this q.

File: compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/Passes.td
Comment (Abhishek-Varma) [inline compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/Passes.td:485]: Nice!

Good to put this as PR's description.

**Review by Abhishek-Varma — APPROVED** [review]
LGTM. Approving to unblock - a few comments to address before merging this in.

File: compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/AMDAIERemoveMemorySpace.cpp
Comment (jtuyls) [reply compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/AMDAIERemoveMemorySpace.cpp:139]: Ok, I am getting rid of this op, so soon we can just delete these lines. Not worth investigating a more 'sustainable' way of doing this at this point.

**Review by jtuyls — APPROVED** [review]
LGTM

**Pull request merged**
